# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/explicit-resource-management/syntax.js
# Commit: c7f4b0bfc0a1
# Full Hash: c7f4b0bfc0a191a876fcdfdfff47791be03e46c5
# Author: Debadree Chatterjee <debadree333@gmail.com>
# Date: 2024-12-06 09:28:31
# Description:
#   Bug 1935283 - SyntaxError on using decls without binding list. r=arai
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231279
# ==============================================================================

diff -r 0731fc0f120c -r c7f4b0bfc0a1 js/src/jit-test/tests/explicit-resource-management/syntax.js
--- a/js/src/jit-test/tests/explicit-resource-management/syntax.js	Thu Dec 05 17:42:12 2024 +0000
+++ b/js/src/jit-test/tests/explicit-resource-management/syntax.js	Thu Dec 05 17:56:42 2024 +0000
@@ -102,6 +102,8 @@
 assertThrowsInstanceOf(() => Reflect.parse("for (;;) using x = 10;"), SyntaxError);
 assertThrowsInstanceOf(() => Reflect.parse("foo: using x = 10;"), SyntaxError);
 assertThrowsInstanceOf(() => Reflect.parse("export using x = 10", { target: "module" }), SyntaxError);
+assertThrowsInstanceOf(() => Reflect.parse("{ using a }"), SyntaxError);
+assertThrowsInstanceOf(() => Reflect.parse("{ using a, b }"), SyntaxError);
 
 // Invalid `await using` syntaxes
 assertThrowsInstanceOf(() => Reflect.parse("{ await using }"), SyntaxError);
@@ -122,6 +124,8 @@
 assertThrowsInstanceOf(() => Reflect.parse("export await using x = 10", { target: "module" }), SyntaxError);
 assertThrowsInstanceOf(() => Reflect.parse("for (await using [x,y] of z) {}", { target: "module" }), SyntaxError);
 assertThrowsInstanceOf(() => Reflect.parse("for (await using {x,y} of z) {}", { target: "module" }), SyntaxError);
+assertThrowsInstanceOf(() => Reflect.parse("{ await using a }", { target: "module" }), SyntaxError);
+assertThrowsInstanceOf(() => Reflect.parse("{ await using a, b }", { target: "module" }), SyntaxError);
 
 // Valid usage of `using` syntax with contextual keywords
 Reflect.parse("{ using await = {}; }");
